src/emission_profile.jl

```diff
@@ -0,0 +1,82 @@
+#imports
+using Plots
+
+#functions
+"""
+From Reynolds (2020) (Eq2-4)
+Calculates the radius of the inner most stable circular orbit of the a black hole,
+with spin a_star, and mass M.
+"""
+function r_isco(a_star, M)
+    
+    a = a_star
+    r_g = 2*G*M/(c^2)
+    z1 = 1+∛(1-a^2)*(∛(1+a)+∛(1-a))
+    z2 = √(3*a^2+z1^2)
+    if a >= 0
+        r_isco = (3+z2-(3-z1)*√(3+z1+2*z2))*r_g
+    elseif a < 0
+        r_isco = (3+z2+(3-z1)*√(3+z1+2*z2))*r_g
+    end
+end
+
+"""
+From Page & Thorne (1974) (Eq15n)
+Calculates the function f at radius r, for a black hole of spin a_star, and mass M, 
+which is used to calculate the flux given by its accretion disk.
+"""
+function f(r, a_star, M)
+
+    R_isco = r_isco(a_star, M)
+  
+    x = √(r/M)
+    x0 = √(R_isco/M)
+    x1 = 2*cos((1/3)*(acos(a_star))-(π/3))
+    x2 = 2*cos((1/3)*(acos(a_star))+(π/3))
+    x3 = -2*cos((1/3)*(acos(a_star)))
+
+    f = (3/(2*M))*(1/(x^2*(x^3-(3*x)+(2*a_star))))*(x-x0-((3/2)*a_star*log(x/x0)) 
+            - (3*(x1-a_star)^2)/(x1*(x1-x2)*(x1-x3))*log((x-x1)/(x0-x1)) 
+            - (3*(x2-a_star)^2)/(x2*(x2-x1)*(x2-x3))*log((x-x2)/(x0-x2)) 
+            - (3*(x3-a_star)^2)/(x3*(x3-x1)*(x3-x2))*log((x-x3)/(x0-x3)))
+end
+
+function diss(mdot, r, a_star, M)
+    diss = mdot*f(r, a_star, M)/(4*π*r)
+end
+
+function mdot(M)
+    L_edd = 3e4*L_☼*(M/M_☼)
+    Mdot = -L_edd/(c^2*η)
+    mdot = 0.1*Mdot
+end
+
+function temp(r, a_star, M)
+    m_dot = mdot(M)
+    temp = (diss(m_dot, r, a_star, M)/σ_SB)^(1/4)
+end
```

Comment:
For me, this function caused issues, and I am really not sure why, but giving it a slightly different name seemed to fix it. I don't know if that's the same for you? Maybe there was something strange with my environment.

Reply:
We noticed something similar, it seems like Julia sometimes gets a bit confused when there are variables and functions with the same names, saying something along the lines of "error with redefining constant". I think for now we'll try and name all of our functions something like "name_f" just to prevent this for now.

Reply:
I'd have to read the scoping rules for Julia again to be absolutely sure what's going on here, but the temporary fix you propose sounds fine for now! We can easily refactor later.